doc/rst/language/spec/interoperability.rst

```diff
@@ -299,8 +299,27 @@ This type could be referred to within a Chapel program using
 
 and defined by supplying ``foo.h`` on the ``chpl`` command line.
 
+The same applies for a C union. An example would be such:
+
+.. code-block:: chapel
+
+       typedef union _someUnion {
+         float x;
+         double y;
+       } _someUnion;
+
+and this type could be referred to within a Chapel program using
+
+
+.. code-block:: chapel
+
+      extern record _someUnion {
+        var x: real(32);
+        var y: real(64);
+      }
```

Comment:
Sorry I missed this before:  I don't think the code above is incorrect, but I think it would be clearer if the typedef'd name of the union was `someUnion` rather than `_someUnion` to be more symmetrical with the struct case.  This would also require the `extern record` declaration to refer to `someUnion`.  Not a show-stopper, but if you had time to fix it, that'd be good.  Meanwhile I'm reading through the rest of the changes once more and looking at the tests...

Reply:
To be super-clear, I'd use:

```chapel
       typedef union _someUnion {
         float x;
         double y;
       } someUnion;
```

(that is, keep the `_` in the first case, not in the second.  Though this probably reflects a circa 1990's naming convention...)

Reply:
Makes sense, I'll remove the underscores.